src/Application.vala

```diff
@@ -184,6 +189,12 @@ public class Mail.Application : Gtk.Application {
 
         main_window.present ();
     }
+
+    private void check_theme () {
```

Comment:
rather than storing the Settings object.
```suggestion
    private void check_theme (Object obj, ParamSpec pspec) {
        unowned var gtk_settings = (Gtk.Settings) obj;
```

Reply:
ignore me, this would break the call outside of the signal. 🤦🏼

Reply:
Haha sorry I was gonna look at this again but that was my first thought as well 😅